src/node_api.cc

```diff
@@ -2766,7 +2766,26 @@ class Work {
     Work* work = static_cast<Work*>(req->data);
 
     if (work->_complete != nullptr) {
-      work->_complete(work->_env, ConvertUVErrorCode(status), work->_data);
+      napi_env env = work->_env;
+
+      // Establish a handle scope here so that every callback doesn't have to.
+      // Also it is needed for the exception-handling below.
+      v8::HandleScope scope(env->isolate);
+
+      work->_complete(env, ConvertUVErrorCode(status), work->_data);
```

Comment:
If I understand correctly it would be less about convenience and more about safety.  If javascript is execute is should be in the state/context that MakeCallback provides.  If we enforce that, it is safer than documenting and asking the implementer to make sure they do that.